src/search/sql_parser.h

```diff
@@ -41,7 +41,11 @@ struct NumericAtomExpr : WSPad<sor<NumberOrParam, Identifier>> {};
 struct NumericCompareOp : sor<string<'!', '='>, string<'<', '='>, string<'>', '='>, one<'=', '<', '>'>> {};
 struct NumericCompareExpr : seq<NumericAtomExpr, NumericCompareOp, NumericAtomExpr> {};
 
-struct BooleanAtomExpr : sor<HasTagExpr, NumericCompareExpr, WSPad<Boolean>> {};
+struct VectorCompareOp : sor<string<'<', '-', '>'>> {};
+struct VectorCompareExpr : seq<WSPad<Identifier>, VectorCompareOp, WSPad<StringOrParam>> {};
```

Comment:
I'm wondering if we can introduce a new syntax of vector literals here, e.g.
```c++
struct VectorLiteral : seq<WSPad<one<'['>>, Number, star<seq<WSPad<one<','>>>, Number>, WSPad<one<']'>>> {};
```

And in the transformer you can collect all children node (typed Number) inside the VectorLiteral to construct a `std::vector<double>`

Reply:
This looks good to me. Perhaps supporting both text and binary formats for SQL **AND** Redis would be more flexible, allowing users to choose the format that best suits their needs?

Reply:
It looks good, but I personally prefer not to use binary format in the syntax because it is more suited for programming interfaces and is not easily readable.

Reply:
I also personally prefer to only support text, but we need to document it in the future, as this is different from Redis vector search which only supports binary format. 
Will implement `VectorLiteral` and release the next revision very soon!

Reply:
For RediSearch part, we need to be compatible to the original implementation in Redis Stack. (which means we only need to support vector parameters in binary format, without any custom syntax).

But for SQL part, we can add our own syntax while try to be similar to the syntax of MySQL. (so that we can support both vector parameters in binary format and also vector literals).

There are many bad design in RediSearch and we can avoid them in SQL part (e.g. no vector literals).

Reply:
I see I see, finally get your point, thanks a lot!!

Reply:
Also for the documentation, yeah we really need a separate documentation page for users to learn how to use Kvrocks Search and what the difference is between RediSearch and Kvrocks Search.

Reply:
I could probably help with that after getting done most part of vector search, will take note of that on the tracking issues.